examples/ecs/system_sets.rs

```diff
@@ -0,0 +1,163 @@
+/// Physics                     (Criteria: App has run < 1.0 seconds)
+///     \--> update_velocity        (via label PhysicsSystem::UpdateVelocity)
+///     \--> movement               (via label PhysicsSystem::Movement)
+/// PostPhysics                 (Criteria: Resource `done` is false)
+///     \--> collision || sfx
+/// Exit                        (Criteria: Resource `done` is true)
+///     \--> exit
+/// ```
+///
+/// The `Physics` label represents a [SystemSet] containing two systems.
+/// This set's criteria is to stop after a second has elapsed.
+/// The two systems (update_velocity, movement) runs in a specified order.
+///
+/// Another label `PostPhysics` uses run criteria to only run after `Physics` has finished.
+/// This set's criteria is to run only when _not done_, as specified via a resource.
+/// The two systems here (collision, sfx) are not specified to run in any order, and the actual
+/// ordering can then change between invocations.
+///
+/// Lastly a system with run criterion _done_ is used to exit the app.
+/// ```
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<Done>()
+        // Note that the system sets added in this example sets their run criteria explicitly.
+        // See the `ecs/state.rs` example for a pattern where run criteria are set implicitly for common
+        // use cases- typically state transitions.
+        // Also note that a system set has a single run criterion at most, which means using `.with_run_criteria(...)`
+        // after `SystemSet::on_update(...)` would override the state transition criterion.
+        .add_system_set(
+            SystemSet::new()
+                // This label is added to all systems in this set.
+                // The label can then be referred to elsewhere (other sets).
+                .label(Physics)
+                .with_system(
+                    update_velocity
+                        .system()
+                        // Only applied to the `update_velocity` system
+                        .label(PhysicsSystem::UpdateVelocity),
+                )
+                .with_system(
+                    movement
+                        .system()
+                        // Only applied to the `movement` system
+                        .label(PhysicsSystem::Movement)
+                        // Enforce order within this system by specifying this
+                        .after(PhysicsSystem::UpdateVelocity),
+                )
+                // This criteria ensures this whole system set only runs when this system's
+                // output says so (ShouldRun::Yes)
+                .with_run_criteria(run_for_a_second.system()),
+        )
+        .add_system_set(
+            SystemSet::new()
+                .label(PostPhysics)
+                // `collision` and `sfx` are not ordered with respect to
+                // each other, and may run in any order
+                .with_system(collision.system())
```

Comment:
Small nit: I like to define systems last in system sets as a general pattern. This creates a nice `SetConfig -> Systems` order.

Reply:
Ok! I'm all for consistency. Updated.